dynatrace_extension/cli/create/extension_template/setup.py.template

```diff
@@ -1,27 +1,196 @@
+import json
 from pathlib import Path
+import re
 from setuptools import setup, find_packages
 
 
-def find_version() -> str:
-    version = "0.0.1"
-    extension_yaml_path = Path(__file__).parent / "extension" / "extension.yaml"
-    try:
-        with open(extension_yaml_path, encoding="utf-8") as f:
-            for line in f:
-                if line.startswith("version"):
-                    version = line.split(" ")[-1].strip("\"")
-                    break
-    except Exception:
-        pass
-    return version
-
-
-setup(name="%extension_name%",
-      version=find_version(),
-      description="%Extension_Name% python EF2 extension",
-      author="Dynatrace",
+class ExtensionMetadata():
+    '''
+    Collects and makes available all extension metadata available to populate
+    the setup() command when building the extension.
+    '''
+
+    def __init__(self):
+        '''
+        Initialize our internal data structures accordingly.
+        '''
+
+        self._activation = self.__read_activation()
+        self._extension = self.__read_extension()
+
+    def __read_activation(self):
+        '''
+        Load the metadata for this extension from the activationSchema.json
+        file. If a failure occurs (i.e. no file), initialize with an empty
+        dict so that subsequent calls just return the default instead.
+
+        Returns:
+            dict containing configuration from schemaActivation.json.
+        '''
+
+        try:
+            with open(Path(__file__).parent / "extension" / "activationSchema.json",
```

Comment:
I don't think this is a good idea, this file can be called anything and some extensions have more than one activation schema

Reply:
interesting point.  I did not realize this was the case (multiple activationSchema.json).  I will take this into consideration.